meraki_appliance.tf

```
@@ -889,7 +912,15 @@ locals {
               fail_over_criterion            = try(vpn_traffic_uplink_preference.fail_over_criterion, local.defaults.meraki.domains.organizations.networks.appliance.traffic_shaping.uplink_selection.vpn_traffic_uplink_preferences.fail_over_criterion, null)
               performance_class_type         = try(vpn_traffic_uplink_preference.performance_class.type, local.defaults.meraki.domains.organizations.networks.appliance.traffic_shaping.uplink_selection.vpn_traffic_uplink_preferences.performance_class.type, null)
               builtin_performance_class_name = try(vpn_traffic_uplink_preference.performance_class.builtin_performance_class_name, local.defaults.meraki.domains.organizations.networks.appliance.traffic_shaping.uplink_selection.vpn_traffic_uplink_preferences.performance_class.builtin_performance_class_name, null)
-              custom_performance_class_id    = try(vpn_traffic_uplink_preference.performance_class.custom_performance_class_id, local.defaults.meraki.domains.organizations.networks.appliance.traffic_shaping.uplink_selection.vpn_traffic_uplink_preferences.performance_class.custom_performance_class_id, null)
+              custom_performance_class_id = try(
+                local.custom_performance_class_ids_by_name[
+                  try(
+                    vpn_traffic_uplink_preference.performance_class.custom_performance_class_name,
+                    local.defaults.meraki.domains.organizations.networks.appliance.traffic_shaping.uplink_selection.vpn_traffic_uplink_preferences.performance_class.custom_performance_class_name
+                  )
+                ],
+                null
+              )
```

```suggestion
              custom_performance_class_id = try(meraki_appliance_traffic_shaping_custom_performance_class.networks_appliance_traffic_shaping_custom_performance_classes[format("%s/%s/%s/%s", domain.name, organization.name, network.name, vpn_traffic_uplink_preference.performance_class.custom_performance_class_name)].id, null)
```
This way:
- the whole key is used, so the same name can be used in e.g. different networks;
- the default is not applied - I don't know if referring to a resource that is user-defined in a default makes sense (also, we haven't been applying defaults in name references so far);
- the extra variable is not used (I would prefer to use ID mapping variables only in special cases - right now, only `organization_ids` and `networks_switch_stacks_routing_interface_ids`).